scipy/optimize/tests/test__differential_evolution.py

```diff
@@ -703,6 +703,7 @@ def f(x):
         x_opt = (1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 3, 3, 1)
         f_opt = -15
 
+        assert_allclose(f(x_opt), f_opt)
```

Comment:
Did you mean to do this? f_opt, x_opt, and f are all hardcoded. This checks that test itself was coded consistently but doesn't test `differential_evolution`, right?

Reply:
One of the costly lessons I've experienced in life is that it's always good to do this kind of check to ensure that the cost function is being calculated correctly.

Reply:
Ok. I checked this for all the problems, of course, but if you're concerned that the test might inadvertently be modified I understand.

Reply:
When I wrote the >120 benchmark functions for examining global optimisers I used the x_opt and f_opt from original source papers. More frequently than not `f(x_opt) != f_opt`. Indeed, a lot of the time f_opt wasn't even the most optimal.

Reply:
Huh! Well, all of these ones seemed to match, and DE seemed to approach them. I manually polished most of them with trust_constr and I think it got slightly better numbers on one, but the improvement was small compared to the difference between the variation in DE's solution, so I didn't really take note.